OpenUpSA/wazimap-ng-ui
# Feat/add cypress integration

## Description

Add Cypress Integration and eventually E2E testing. 
Reuse the BDD features that were already created and use a library to help glue cypress and BDD together. 

Runs on PRs against netlify PR review apps and on pushes to staging against staging.

## Related Issue
_no issue created_

## How to test it locally
run the local server with `npm start` and then run `npm run cypress:open` in a separate process to see it in-real time :)

## Screenshots
<img width="2041" alt="Bildschirmfoto 2020-09-16 um 15 15 13" src="https://user-images.githubusercontent.com/688980/93342353-882ca500-f82f-11ea-825b-9561ffa7d221.png">
<img width="929" alt="Bildschirmfoto 2020-09-16 um 15 14 41" src="https://user-images.githubusercontent.com/688980/93342378-8f53b300-f82f-11ea-9b7e-7c3bd3729da2.png">


## Changelog

### Added
* cypress configuration
* cypress-cucumber (to reuse the BDD feature tests)
* cypress-testing-library (for easier finding of dom elements)
* gh action to run on PRs
* gh action to run on push to staging

### Updated
* copied one feature test to cypress

### Removed


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
